### PR TITLE
GoogleStackdriverTarget - Extracted RenderJsonPayload from BuildLogEntry

### PR DESCRIPTION
**Pull request by snakefoot**
Simple refactoring, no changes in logic.

### CODE REVIEW

**Comment by amanda-tarafa** [comment]
Thanks for this. I haven't had the time to look at it today, but I'll do that first thing tomorrow morning.

File: apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs [hunk 1]
```diff
@@ -514,66 +517,19 @@ private LogEntry BuildLogEntry(LogEventInfo loggingEvent)
...
                 logEntry.TextPayload = RenderLogEvent(Layout, loggingEvent);
 
-                foreach (var combinedProperty in GetAllProperties(loggingEvent).Where(x => !string.IsNullOrEmpty(x.Key)))
+                foreach (var combinedProperty in GetAllProperties(loggingEvent))
                 {
+                    if (string.IsNullOrEmpty(combinedProperty.Key))
+                    {
+                        continue;
+                    }
+
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs:532]: Hmmm, I like the old way better. I think it's more readable.
Any particular reason you'd prefer this?

File: apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs [hunk 2]
```diff
@@ -598,10 +554,72 @@ private LogEntry BuildLogEntry(LogEventInfo loggingEvent)
...
+                var propertiesStruct = new Struct();
+                jsonStruct.Fields.Add("properties", Value.ForStruct(propertiesStruct));
+
+                foreach (var combinedProperty in GetAllProperties(loggingEvent))
+                {
+                    if (string.IsNullOrEmpty(combinedProperty.Key))
+                    {
+                        continue;
+                    }
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs:602]: Same comment/question as before.

File: apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs [hunk 3]
```diff
@@ -598,10 +554,72 @@ private LogEntry BuildLogEntry(LogEventInfo loggingEvent)
...
+
+                    Value jsonValue;
+                    try
+                    {
+                        jsonValue = _jsonConvertFunction(combinedProperty.Value);
+                    }
+                    catch (Exception ex)
+                    {
+                        InternalLogger.Warn(ex,
+                            "GoogleStackdriver(Name={0}): Exception at BuildLogEntry with Key={1}", Name, combinedProperty.Key);
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs:612]:
```suggestion
                            "GoogleStackdriver(Name={0}): Exception at RenderJsonPayload with Key={1}", Name, combinedProperty.Key);
```

**Review by amanda-tarafa — CHANGES_REQUESTED** [review]
One tiny change and a question. Thanks!

File: apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs (hunk 3 above)
Comment (snakefoot) [reply apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs]: Good catch.

File: apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs (hunk 1 above)
Comment (snakefoot) [reply apis/Google.Cloud.Logging.NLog/Google.Cloud.Logging.NLog/GoogleStackdriverTarget.cs:532]: I really like performance and calling any Linq-expression besides `Any()` or `Count()` performs bonus-class-allocation, even when the collection is empty.

**Comment by amanda-tarafa** [comment]
The diff checker is failing but it shouldn't. I'll try to take a look later today, else tomorrow morning.

**Comment by snakefoot** [comment]
Not sure what this error means:

```
Checking compatibility for Google.Cloud.DocumentAI.V1 version 1.0.0-beta01
Differences from 1.0.0-beta01
Failed: System.IO.DirectoryNotFoundException: Could not find a part of the path '/home/runner/work/google-cloud-dotnet/google-cloud-dotnet/apis/Google.Cloud.DocumentAI.V1/Google.Cloud.DocumentAI.V1/bin/Release/netstandard2.0/Google.Cloud.DocumentAI.V1.dll'.
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.FileStream.OpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   at System.IO.File.ReadAllBytes(String path)
   at Google.Cloud.Tools.VersionCompat.Assemblies.LoadFile(String file) in /home/runner/work/google-cloud-dotnet/google-cloud-dotnet/tools/Google.Cloud.Tools.VersionCompat/Assemblies.cs:line 122
   at Google.Cloud.Tools.ReleaseManager.CheckVersionCompatibilityCommand.CheckCompatibility(ApiMetadata api, StructuredVersion version) in /home/runner/work/google-cloud-dotnet/google-cloud-dotnet/tools/Google.Cloud.Tools.ReleaseManager/CheckVersionCompatibilityCommand.cs:line 140
   at Google.Cloud.Tools.ReleaseManager.CheckVersionCompatibilityCommand.Execute(String[] args) in /home/runner/work/google-cloud-dotnet/google-cloud-dotnet/tools/Google.Cloud.Tools.ReleaseManager/CheckVersionCompatibilityCommand.cs:line 84
   at Google.Cloud.Tools.ReleaseManager.Program.Main(String[] args) in /home/runner/work/google-cloud-dotnet/google-cloud-dotnet/tools/Google.Cloud.Tools.ReleaseManager/Program.cs:line 56
```

**Comment by snakefoot** [comment]
Closing and reopening to check the stability of the cloud.

**Comment by amanda-tarafa** [comment]
Don't worry about the diff checker, it's something on our side, not the PR. I'll take a look tomorrow.

**Comment by amanda-tarafa** [comment]
@snakefoot Your branch is some commits behind ours, can you rebase and push the PR again. It should take care of the diff checker errors.
Thanks

**Comment by snakefoot** [comment]
Ok strange. Seems the issue arrived with applying your suggested change from GitHub-review. Better not use that feature


Sent from my Sony Xperia


---- Amanda Tarafa Mas wrote ----


@snakefoot<https://github.com/snakefoot> Your branch is some commits behind ours, can you rebase and push the PR again. It should take care of the diff checker errors.
Thanks

**Comment by snakefoot** [comment]
Rebased and force-pushed back to apply the github-review-suggestion

**Review by amanda-tarafa — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**